node/src/components/contract_runtime/operations.rs

```diff
@@ -360,73 +463,66 @@ pub fn execute_finalized_block(
             }
             FeeHandling::PayToProposer => {
                 // in this mode, the consumed gas is paid as a fee to the block proposer
-                let consumed = Gas::new(artifact_builder.consumed());
-                let handle_payment_request = HandlePaymentRequest::new(
+                let amount = cost.saturating_sub(refund_amount);
```

Comment:
Not sure this is correct. If the transaction is using standard payment, the `refund_amount` here will be 0. Which means that the full cost goes from the initiator to the proposer. Shouldn't we apply the ratio to the cost?

Reply:
Yes, I'll add a handle refund option to just get the amount.